### ISSUE TITLE
Unable to shutdown bot

### ISSUE THREAD

**Issue by davfsa** [nekokatt/hikari#113]
### Steps to reproduce
1. Start bot up
2. Stop the bot with `KeyboardInterrupt`

### Expected Result
Bot shuts down

### Actual Result
Bot stays online

### Traceback
```py
  oooo         o8o  oooo                            o8o       光 v2.0.0 [HEAD]
  `888         `"'  `888                            `"'       © 2020 Nekokatt, licensed under MIT
   888 .oo.   oooo   888  oooo   .oooo.   oooo d8b oooo       CPython 3.8.2
   888P"Y88b  `888   888 .8P'   `P  )88b  `888""8P `888       running on:    5.4.0-45-generic Linux x86_64
   888   888   888   888888.     .oP"888   888      888       Installed at:  /home/davfsa/coding/test-bots/hikari
   888   888   888   888 `88b.  d8(  888   888      888       Documentation: https://nekokatt.github.io/hikari
  o888o o888o o888o o888o o888o `Y888""8o d888b    o888o      Support:       https://discord.gg/Jx4cNGG

I 2020-09-02 08:59:28,130 hikari: this application is stateless, cache-based operations will not be available
I 2020-09-02 08:59:28,693 hikari: single-sharded configuration -- you have started 0/1000 sessions prior to connecting (resets at 02/09/20 08:59:28 CEST)
I 2020-09-02 08:59:28,997 hikari.gateway.0: received HELLO, heartbeat interval is 41.25s
I 2020-09-02 08:59:29,201 hikari.gateway.0: shard is ready [session:53d3a6ac33cbf7fb2095bb1c43546588, user_id:649186216436039680, tag:Hikari Testing#6464]
^CI 2020-09-02 09:07:42,225 hikari: received signal to shut down client
I 2020-09-02 09:07:42,226 hikari.gateway.0: received request to shut down shard
I 2020-09-02 09:07:42,228 hikari.gateway.0: gateway client closed, will not attempt to restart
I 2020-09-02 09:07:42,441 hikari.gateway.0: received HELLO, heartbeat interval is 41.25s
W 2020-09-02 09:07:42,551 hikari.gateway.0: invalid session, so will attempt to reconnect with new session in a few seconds
I 2020-09-02 09:07:46,388 hikari.gateway.0: received HELLO, heartbeat interval is 41.25s
I 2020-09-02 09:07:46,677 hikari.gateway.0: shard is ready [session:1b4578f4850c678c90c88dbec76e6817, user_id:649186216436039680, tag:Hikari Testing#6464]
# Me trying to shut it down again
^CI 2020-09-02 09:07:47,754 hikari: received signal to shut down client
^CI 2020-09-02 09:07:49,105 hikari: received signal to shut down client
^CI 2020-09-02 09:07:49,552 hikari: received signal to shut down client
^CI 2020-09-02 09:07:49,754 hikari: received signal to shut down client
^CI 2020-09-02 09:07:58,969 hikari: received signal to shut down client
^CI 2020-09-02 09:07:59,137 hikari: received signal to shut down client
^CI 2020-09-02 09:07:59,312 hikari: received signal to shut down client
^CI 2020-09-02 09:07:59,475 hikari: received signal to shut down client
^CI 2020-09-02 09:07:59,625 hikari: received signal to shut down client
^CI 2020-09-02 09:07:59,799 hikari: received signal to shut down client
^CI 2020-09-02 09:07:59,985 hikari: received signal to shut down client
^CI 2020-09-02 09:08:00,151 hikari: received signal to shut down client
```

### Version info
hikari v2.0.0 HEAD
located at /home/davfsa/coding/hikari/hikari
CPython 3.8.2 GCC 9.3.0
Linux carbono 5.4.0-45-generic #49-Ubuntu SMP Wed Aug 26 13:38:52 UTC 2020 x86_64 x86_64

### Further info
I have not been able to reproduce this again. Maybe we should suppress all other closures when we receive the signal to shutdown the bot?

### Checklist

- [x] I have made sure to remove ANY sensitive information (bot token, passwords,
      credentials, personal details, etc).
- [x] I have searched the issue tracker and have made sure it's not a duplicate.
      If it is a follow up of another issue, I have specified it.

**Comment by nekokatt** [comment]
Invalid session is caused by you starting the bot more than once in 5 seconds.

**Comment by davfsa** [comment]
Will edit comment

**Comment by nekokatt** [comment]
This has mostly been fixed as of 2.0.0.dev75
